src/modules/vpaid.js

```diff
@@ -501,6 +501,22 @@ export default function (playerInstance, options) {
 
         }, 100);
 
+        // Handles updates for generic iframes
+        vpaidIframe.contentWindow.addEventListener('message', (event) =>  {
+            const parsedEvent = (() => {
+                try { return JSON.parse(event.data); }
+                catch (e) { /* do nothing */ }
+            })();
+
+            if (
+                parsedEvent &&
+                parsedEvent.event === 'infoDelivery' &&
```

Comment:
Where is this message generated? I don't see in VPAID doc anything about any infoDelivey thing.
Is this something specific to the actual one ad that is being tested from somewhere? That would be a big red flag.

Reply:
This message is being generated by the VPAID ad iframe content, it was implemented to simulate behavior found with the same VPAID tag on `https://googleads.github.io/googleads-ima-html5/vsi/` as it's reading the same message to track the passage of time. It's indeed something specific to the ad tested, any recommendation on how to handle this?

Reply:
I don't even think infoDelivery is being generated by VPAID. It's being generated specifically by YouTube (https://stackoverflow.com/questions/65511523/how-to-listen-to-time-change-events-with-the-youtube-iframe-player-api).
What if there is some other streaming plaform used? 
Anyway... This is NOT the way to do it. 
The https://googleads.github.io/googleads-ima-html5/vsi/ uses google SDK. The code of that SDK doen't have a single reference to infoDelivery.. Istead it uses some abstractions defined in here (AdDurationChange, AdLinearChange, etc.) :
https://www.iab.com/wp-content/uploads/2015/06/VPAID_2_0_Final_04-10-2012.pdf

I would pay attention to these messages:
![Screenshot 2021-12-07 at 09 55 43](https://user-images.githubusercontent.com/3126345/145008321-a15dcdb2-ade7-4b50-ba84-d91a8e4953e5.png)
![Screenshot 2021-12-07 at 09 59 02](https://user-images.githubusercontent.com/3126345/145009655-41b0bb53-6b20-41c0-88d6-972934b0cdbd.png)
![Screenshot 2021-12-07 at 10 04 20](https://user-images.githubusercontent.com/3126345/145009669-ba52b47c-723c-4357-ab00-909069d13684.png)

You would notice that the vpaid script loaded from the advertiser implements those abstractions (ad the VPAID script itself KNOWS what player it uses, so it in turn does use infoDelivery:
https://bestonlinevideo.co/Campaigns/2020/VPaid_V2/VPaid_V3.8.6.js
We are not supposed to though.

We are supposed to be using the events defined in VAST document. The ones I mentioned, like AdLinearChange. This is the correct generic approach.